### PR TITLE
Favourites

### PR DESCRIPTION
**Pull request by jaggu4444**
Favourites feature added to list the favourite files/folders in favourites page in owncloud.
But, to run the test error freely, PHP code for this should be written.
Please do have a look @individual-it , @phil-davis

### CODE REVIEW

**Comment by individual-it** [comment]
please make a PR with only the commit that belongs to that PR. In this case "test for marking file and folder as favourite"
please make sure always branch off from master

**Comment by jaggu4444** [comment]
I think I have corrected it. Please do have a look @phil-davis

File: tests/ui/features/bootstrap/FilesContext.php [hunk 1]
```diff
@@ -897,6 +918,32 @@ public function theContentOfShouldNotHaveChanged($fileName) {
 		$this->assertContentOfRemoteAndLocalFileIsSame($remoteFile, $localFile);
 	}
 
+	/**
+	 * @When I mark the file/folder :title as favorite
+	 * @param string $title
+	 * @return void
+	 */
+	public function iMarkTheFileAsFavorite($title) {
```
Comment (individual-it) [inline tests/ui/features/bootstrap/FilesContext.php]: I don't like that `$title` variable name

File: tests/ui/features/bootstrap/FilesContext.php [hunk 2]
```diff
@@ -897,6 +918,32 @@ public function theContentOfShouldNotHaveChanged($fileName) {
...
+		$findRow = $this->filesPage->findFileRowByName($title, $this->getSession());
```
Comment (individual-it) [inline tests/ui/features/bootstrap/FilesContext.php]: `$findRow` or rather `$fileRow`?

File: tests/ui/features/bootstrap/FilesContext.php [hunk 3]
```diff
@@ -897,6 +918,32 @@ public function theContentOfShouldNotHaveChanged($fileName) {
...
+		$findRow->markAsFavorite();
+		$this->filesPage->waitTillFileRowsAreReady($this->getSession());
+	}
+	
+	/**
+	 * @Then the file/folder :title should be marked as favorite
+	 * @param string $title
+	 * @return void
+	 */
+	public function theFileShouldBeMarkedAsFavorite($title) {
```
Comment (individual-it) [inline tests/ui/features/bootstrap/FilesContext.php]: same as above

File: tests/ui/features/lib/FavoritesPage.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * ownCloud
+ *
+ * @author Artur Neumann <artur@jankaritech.com>
```
Comment (individual-it) [inline tests/ui/features/lib/FavoritesPage.php]: have I created that file? I cannot remember ;-)

File: tests/ui/features/lib/FilesPageElement/FileRow.php [hunk 5]
```diff
@@ -289,4 +291,28 @@ public function findFileLink() {
...
 		$this->findFileLink()->click();
 	}
+
+	/**
+	 * marks the current file or folder as favorite by clicking the star icon
+	 *
+	 * @return void
+	 */
+	public function markAsFavorite() {
+		$this->rowElement->find("xpath", $this->notMarkedFavoriteXpath)->click();
```
Comment (individual-it) [inline tests/ui/features/lib/FilesPageElement/FileRow.php]: we need a check to see if `find()` returns `null`

**Comment by individual-it** [comment]
please also rebase on master and resolve conflicts

**Pull request closed without merging**